### DISCUSSION TITLE
guide-R/analyses/contrastes

### DISCUSSION THREAD

**Discussion by giscus** [larmarange/guide-R#22] [Forum]
# guide-R/analyses/contrastes

https://larmarange.github.io/guide-R/analyses/contrastes.html

**Comment by KMATSOU12** [comment]
Bonjour Larmarange, 

Merci beaucoup pour votre blog qui est une véritable ressource. 
J'ai une question sur les contrasts. 
Pour contrsum=TRUE, serait-il possible d'avoir en plus de l'estimate, l'intervalle de confiance et la p.value associée avec l'argument add_estimate_to_reference_rows = TRUE ?
Je sais que c'est possible de ruser en changeant les références de la variable et récupérer la ligne manquante (estimate, 95%CI et P.value). 
Merci une fois de plus pour votre réponse.

**Reply by larmarange** [reply]
Bonjour,

en effet, ce n'est actuellement pas le cas avec `tbl_regression()`

Je viens d'ajouter la fonctionnalité à `broom.helpers` qui est utilisé par `gtsummary` pour produire le tableau de résultats. Pour utiliser cette version de développement, il faut utiliser la commande `remotes::install_github("larmarange/broom.helpers#220")`.

Il faudra attendre quelques semaines pour que cela devienne disponible dans la prochaine version stable de `broom.helpers`.

``` r
library(gtsummary)
#> #StandWithUkraine

df <- Titanic %>%
  dplyr::as_tibble() %>%
  dplyr::mutate(dplyr::across(where(is.character), factor))

mod <- df %>%
  glm(
    Survived ~ Class + Age + Sex,
    data = ., weights = .$n, family = binomial,
    contrasts = list(Age = contr.sum, Class = "contr.sum")
  )

mod %>%
  tbl_regression(
    add_estimate_to_reference_rows = TRUE
  ) %>%
  as_kable()
```

| **Characteristic** | **log(OR)** |  **95% CI**  | **p-value** |
|:-------------------|:-----------:|:------------:|:-----------:|
| Class              |             |              |             |
| 1st                |    0.91     |  0.70, 1.1   |   \<0.001   |
| 2nd                |    -0.10    | -0.34, 0.12  |     0.4     |
| 3rd                |    -0.86    | -1.1, -0.67  |   \<0.001   |
| Crew               |    0.06     | -0.17, 0.28  |     0.5     |
| Age                |             |              |             |
| Adult              |    -0.53    | -0.77, -0.29 |   \<0.001   |
| Child              |    0.53     |  0.26, 0.80  |   \<0.001   |
| Sex                |             |              |             |
| Female             |    0.00     |      —       |             |
| Male               |    -2.4     |  -2.7, -2.1  |   \<0.001   |

``` r

mod %>%
  tbl_regression(
    exponentiate = TRUE,
    add_estimate_to_reference_rows = TRUE
  ) %>%
  as_kable()
```

| **Characteristic** | **OR** | **95% CI** | **p-value** |
|:-------------------|:------:|:----------:|:-----------:|
| Class              |        |            |             |
| 1st                |  2.49  | 2.01, 3.10 |   \<0.001   |
| 2nd                |  0.90  | 0.71, 1.13 |     0.4     |
| 3rd                |  0.42  | 0.35, 0.51 |   \<0.001   |
| Crew               |  1.06  | 0.84, 1.33 |     0.5     |
| Age                |        |            |             |
| Adult              |  0.59  | 0.46, 0.75 |   \<0.001   |
| Child              |  1.70  | 1.29, 2.24 |   \<0.001   |
| Sex                |        |            |             |
| Female             |  1.00  |     —      |             |
| Male               |  0.09  | 0.07, 0.12 |   \<0.001   |

<sup>Created on 2023-03-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by ShermanWidenfp2** [reply]
Bonjour,

Je suis désolé, mais je ne parle pas français. Cependant, en utilisant un traducteur en ligne, je comprends que vous avez une question sur les contrasts dans R et que vous souhaitez savoir s'il est possible d'obtenir l'intervalle de confiance et la valeur p associée avec l'argument add_estimate_to_reference_rows = TRUE.

Malheureusement, je ne suis pas suffisamment familier avec cette fonctionnalité pour répondre à votre question. Je vous recommande de consulter la documentation de R ou de poser votre question sur un forum de discussion de R pour obtenir une réponse plus précise.

Je suis désolé de ne pas pouvoir vous aider davantage.
